src/Storages/StorageS3.cpp

```diff
@@ -660,8 +660,19 @@ Pipe StorageS3::read(
     Block block_for_format;
     if (isColumnOriented())
     {
+        auto fetch_columns = column_names;
+        const auto & virtuals = getVirtuals();
+        fetch_columns.erase(
+            std::remove_if(
+                fetch_columns.begin(), fetch_columns.end(),
+                [&](const String & col)
+                { return std::any_of(virtuals.begin(), virtuals.end(), [&](const NameAndTypePair & virtual_col){ return col == virtual_col.name; }); }));
```

Comment:
Just a good thing to know, there is [std::erase_if](https://en.cppreference.com/w/cpp/container/vector/erase2)